### DISCUSSION TITLE
How to configure MiniDexed to show the patch/performance on the LCD, initially and after MIDI Program Change?

### DISCUSSION THREAD

**Discussion by Andy2No** [probonopd/MiniDexed#659] [Q&A] [answered]
I finally got around to adding a 1602 LCD display to MiniDexed.  I was expecting to see the patch/performance name on it, and assumed that would be updated after sending a MIDI Program Change message.

I haven't yet fitted an encoder, and wasn't immediately planning to because I only really wanted to be able to change patches and see the patch name.  I can send Program Change messages easily, and can send Bank Select less easily - because it has to be done as a full Bank Select MSB, Bank Select LSB, PC sequence, which is awkward, but at least I can do that.

As I expect you're all aware, all I currently see on the LCD is:

_______MiniDexed
TG1____________->

Is there a setting to get it to show the current patch or performance name when it starts up, and to update that each time I change it by sending a MIDI PC, etc?

**Comment by probonopd** [comment, marked as answer]
Hello @Andy2No.
You need to use the menu as described in https://github.com/probonopd/MiniDexed/wiki/Menu.

To see e.g., the name of the voice in the first Tone Generator (TG1), when you see

![image](https://github.com/probonopd/MiniDexed/assets/2480569/87a59e66-fb3d-45de-bfee-e18ec3602152)

click the knob to open the submenu for that tone generator. You will see

![image](https://github.com/probonopd/MiniDexed/assets/2480569/344182dd-12e9-462d-bc2c-1aa8922d7912)

Click the knob again to open the "Voice "submenu. You will then see the name of the loaded voice:

![image](https://github.com/probonopd/MiniDexed/assets/2480569/1620fb9a-ca2e-4055-9550-6fc71a5206e1)

Likewise, to see the performance names, you need to go to the Performances section of the menu as explained on the linked page.

**Comment by Andy2No** [comment]
@probonopd Thanks.  I got it working with MIDI buttons first, with a Korg nanoKontrol and a USB-MIDI to DIN MIDI converter (based on a HobbyTronics board that might not be available any more).  I'm using an M-Audio MidiSport instead of the recommended MIDI In circuit, at the moment.  That and using a DAC instead of PWM audio has got rid of the stuck notes problem I was having initially, using an M-Audio Uno.  That was also using an earlier version of MiniDexed, I think, so I'm not entirely sure which of those caused that.

I was a bit confused by the suggested settings for buttons on the Files page.  I wasn't sure if that was how it was meant to be for just an encoder, but having tried it, I see it is.  I could maybe think about how to word that part better, to make it clearer to the likes of me.

Anyway, I now have it working fairly well with a DFRobot EC11 Rotary Encoder Module, without yet adding any capacitors.  I can see it has four little SMD components on one end, so maybe it has some, but unfortunately the link to the schematic is broken on the Wiki for it:

https://wiki.dfrobot.com/EC11_Rotary_Encoder_Module_SKU__SEN0235

The pins are in a different order from the encoder shown on the Hardware page, and it isn't labelled well on the module, so it's necessary to refer to the diagram for it.

It doesn't appear to bounce but does tend to get ignored if I turn it slowly.  Perhaps that's normal though?  It works well enough, so I'll leave it as it is for now.
